### ISSUE TITLE
Access .Ahead and .Behind of the push-remote in the built-in Git segment

### ISSUE THREAD

**Issue by ccjmne** [JanDeDobbeleer/oh-my-posh#5312]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What would you like to see added?

When you work with a pair of **fetch-** and **push-** remotes, it looks like oh-my-posh's built-in Git segment "only" retrieves the Ahead and Behind values for the **fetch-remote**.
It'd be really neat to be able to also visualise how ahead or behind you are from your **push-remote**!

Here's the equivalent "segment" in p10k: https://github.com/romkatv/powerlevel10k/blob/2b7da93df04acd04d84f5de827e5b14077839a4b/gitstatus/gitstatus.prompt.zsh#L58-L66

I could give it a spin and try to submit a PR that introduces this functionality, if you think it could make sense!

Thanks in advance!

**Comment by JanDeDobbeleer** [comment]
@ccjmne I was looking into this yet I fail to find the correct way to get that information. `git status` doesn't seem to show it, not do I find another command that can tell me this.

**Comment by nopeless** [comment]
@JanDeDobbeleer 

I did some code digging and found the source of that variable

https://github.com/romkatv/gitstatus/blob/44504a24b1b999a4f56ff74c75b8215bdcadee1f/gitstatus.plugin.sh

I don't know enough about git to know what is going on here but it seems like there is an efficient implementation of getting vcs information from the git status daemon using IPC

**Comment by JanDeDobbeleer** [comment]
@nopeless yes but that comes from the other library, this is the code that requests that data. Didn't see any git logic in there (or I'm mistaken).

**Comment by nopeless** [comment]
i wrote my comment poorly, it is not about git. i realized that the daemon is from that repository as well. https://github.com/romkatv/gitstatus/tree/master/src

**Comment by JanDeDobbeleer** [comment]
@nopeless this is the actual logic, albeit it uses libgit2 and not standard git calls. It does roughly indicate what needs to happen, which is what I assumed by looking at it previously.

```
const char* ref = git_reference_name(remote->ref);
// Number of commits we are ahead of upstream. Non-negative integer.
resp.Print(CountRange(repo->repo(), ref + "..HEAD"s));
// Number of commits we are behind upstream. Non-negative integer.
resp.Print(CountRange(repo->repo(), "HEAD.."s + ref));
```

**Comment by sanki92** [comment]
@JanDeDobbeleer I've implemented this feature in PR #6851 .

Added two new template properties to the Git segment:
- `.PushAhead` - commits ahead of push-remote
- `.PushBehind` - commits behind of push-remote

The implementation:
- Respects Git's push-remote configuration hierarchy (`branch.pushRemote`, `remote.pushDefault`, fallback to upstream)
- Uses `git rev-list --count` for accurate calculation
- Only executes when `fetch_status` is enabled
- Includes comprehensive tests and documentation

Users can now display push-remote status in their prompt templates, similar to how powerlevel10k handles it.

**Comment by JanDeDobbeleer** [comment]
@sanki92 you're on a roll! Allow me this weekend to get to a review, We're having a birthday at home 😃

**Comment by sanki92** [comment]
> [@sanki92](https://github.com/sanki92) you're on a roll! Allow me this weekend to get to a review, We're having a birthday at home 😃

@JanDeDobbeleer Thank you! Take your time - I'm happy to address any feedback when you're ready. 

I appreciate you taking the time to review these contributions!

**Comment by JamBalaya56562** [comment]
This looks resolved by #6851 (merged), which adds `.PushAhead` / `.PushBehind` to the Git segment via the `fetch_push_status` option. I think this issue can be closed. Thanks!
